pkg/config/security/api_client.go

```
@@ -0,0 +1,335 @@
+	ClientSecret     string                 `yaml:"clientSecret,omitempty" json:"clientSecret,omitempty"`
+	Entitlements     []string               `yaml:"entitlements" json:"entitlements"`
+	Enabled          bool                   `yaml:"enabled" json:"enabled"`
+	OverrideSettings OverrideSettings       `yaml:"overrideSettings,omitempty" json:"overrideSettings,omitempty"`
+	Description      string                 `yaml:"description,omitempty" json:"description,omitempty"`
+	IPFilterOp       string                 `yaml:"ipFilterOp,omitempty" json:"ipFilterOp,omitempty"`
+	IPFilters        []string               `yaml:"ipFilters,omitempty" json:"ipFilters,omitempty"`
+	JWKUri           string                 `yaml:"jwkUri,omitempty" json:"jwkUri,omitempty"`
+	AdditionalConfig AdditionalConfig       `yaml:"additionalConfig,omitempty" json:"additionalConfig,omitempty"`
+	AdditionalProps  map[string]interface{} `yaml:"additionalProperties,omitempty" json:"additionalProperties,omitempty"`
+}
+
+type OverrideSettings struct {
+	RestrictScopes bool    `yaml:"restrictScopes" json:"restrictScopes"`
+	Scopes         []Scope `yaml:"scopes" json:"scopes"`
+}
+
+type Scope struct {
+	Name        string `yaml:"name" json:"name"`
+	Description string `yaml:"description" json:"description"`
+}
+
+type AdditionalConfig struct {
+	ClientAuthMethod                       string   `yaml:"clientAuthMethod" json:"clientAuthMethod"`
+	ValidateClientAssertionJti             bool     `yaml:"validateClientAssertionJti" json:"validateClientAssertionJti"`
+	AllowedClientAssertionVerificationKeys []string `yaml:"allowedClientAssertionVerificationKeys,omitempty" json:"allowedClientAssertionVerificationKeys,omitempty"`
+}
+
+func NewAPIClient() *ApiClient {
+	return &ApiClient{}
+}
+
+func (c *ApiClient) CreateAPIClient(ctx context.Context, apiClientConfig *APIClientConfig) (string, error) {
+	if apiClientConfig == nil {
+		fmt.Println("ERROR: Client object is nil!")
```

`fmt.Println` must never exist. I think this check is not needed in any case. Let the system panic in the event that obvious parameters are not passed in properly.